src/vdaf/prio3.rs

```diff
@@ -252,14 +253,131 @@ pub struct Prio3InputShare<F> {
 
     /// The sum of the joint randomness seed shares sent to the other aggregators.
     //
-    // TODO(cjpatton) If `input.joint_rand_len() == 0`, then we don't need to bother with the
-    // joint randomness seed at all and make this optional.
+    // TODO(cjpatton) If `input.joint_rand_len() == 0`, then we don't need to bother with the joint
+    // randomness seed at all. See https://github.com/cjpatton/vdaf/issues/15.
     pub joint_rand_seed_hint: Key,
 
     /// The blinding factor, used to derive the aggregator's joint randomness seed share.
     pub blind: Key,
 }
 
+#[allow(dead_code)]
+impl<F: FieldElement> Prio3InputShare<F> {
+    pub(crate) fn append(&self, bytes: &mut Vec<u8>) {
```

Comment:
Could this be `append<W>(&self, writer: &mut W) -> Result<(), VdafError>`? [`Vec<u8>` implements `Write`](https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-Write) so it shouldn't be any harder to use for a caller that wants to append to a `Vec` while being more general (maybe I want to blast my `Prio3InputShare` directly into a socket instead of a heap buffer).

Reply:
Yup!